### PR TITLE
WIP: `@deck.gl/gpu-table` module based on AttributeManager related code

### PR DESCRIPTION
**Pull request by ibgreen**
For #3796

#### Change List
-

### CODE REVIEW

File: modules/gpu-table/docs/gpu-table-guide.md [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+table = table.removeColumn('price'); // drops chunks
+gpuTable.setTable(table);
+```
+
+## Transformations
+
+`GPUTable`s are perfect sources for `TransformFeedback` operations. New tables (new columns) can be generated from existing tables by composing or processing existing columns.
+
+```js
+const gpuTable = GPUTransform(gpuTable, {
```
Comment (ibgreen) [inline modules/gpu-table/docs/gpu-table-guide.md]: @1chandu Ideas for transform

Comment (1chandu) [reply modules/gpu-table/docs/gpu-table-guide.md]: Updated with sample implementation.

**Comment by kylebarron** [comment]
> The Apache Arrow type system, does not directly support some key columns types that are commonly used in GPUs, most notably short fixed-size vectors of numbers (mapped to `vec2`, `vec3`, `vec4` etc in GLSL).

This might be a recent Arrow addition, but [Fixed-size Lists](https://arrow.apache.org/docs/format/Columnar.html#fixed-size-list-layout) are now supported:

> A fixed size list type is specified like `FixedSizeList<T>[N]`, where `T` is any type (primitive or nested) and `N` is a 32-bit signed integer representing the length of the lists.

There is some overhead, so I'm not sure whether it's more efficient for short arrays than the current `FixedSizeBinary` + column metadata approach.

**Comment by ibgreen** [comment]
> Fixed-size Lists are now supported:

Astute observation. In fact FixedSizeLists are exactly how we are implementing Arrow support in deck.gl-native prototype. 

They are clearly the way to go. We still need to encode a few optional things as field metadata but we can use arrow types to get gl type and size (number of components) for each attribute.

As you can see, this RFC while (still highly relevant), is outdated and needs a refresh.

**Pull request closed without merging**